### PR TITLE
Simplify returns

### PR DESCRIPTION
**Pull request by carusogabriel**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Using some boolean returns from functions and boolean results, we can simplify some returns :)

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php [hunk 1]
```diff
@@ -45,11 +45,7 @@ class ChoiceFormField extends FormField
     public function hasValue()
     {
         // don't send a value for unchecked checkboxes
-        if (in_array($this->type, array('checkbox', 'radio')) && null === $this->value) {
-            return false;
-        }
-
-        return true;
+        return !(in_array($this->type, array('checkbox', 'radio')) && null === $this->value);
```
Comment (carusogabriel) [inline src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php]: Is there a way to simplify it even more? :thinking:

Comment (GromNaN) [inline src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php]: not(A and B) = not(A) or not(B)

And it seems more logical to write this:
```
return !in_array($this->type, array('checkbox', 'radio')) || null !== $this->value;
```

**Review by GromNaN — APPROVED** [review]

File: src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php [hunk 2]
```diff
@@ -45,11 +45,7 @@ class ChoiceFormField extends FormField
     public function hasValue()
     {
         // don't send a value for unchecked checkboxes
-        if (in_array($this->type, array('checkbox', 'radio')) && null === $this->value) {
-            return false;
-        }
-
-        return true;
+        return !in_array($this->type, array('checkbox', 'radio')) || null !== $this->value;
```
Comment (rugabarbo) [inline src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php]: IMO the original version was more readable.
Expression `<negation> || <negation>` is difficult to understand.

Comment (javiereguiluz) [reply src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php]: I agree! Let's be careful with some of these changes. We must optimize code for humans, not machines. OPCache will compile this into the most optimized code possible.

File: src/Symfony/Component/Form/Test/DeprecationErrorHandler.php [hunk 3]
```diff
@@ -33,11 +33,7 @@ public static function handle($errorNumber, $message, $file, $line, $context)
 
     public static function handleBC($errorNumber, $message, $file, $line, $context)
     {
-        if ($errorNumber & ~E_USER_DEPRECATED) {
-            return true;
-        }
-
-        return false;
+        return $errorNumber & ~E_USER_DEPRECATED;
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Test/DeprecationErrorHandler.php]: this one now returns an integer

File: src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php (hunk 2 above)
Comment (carusogabriel) [reply src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php]: Gonna revert this one :+1:

File: src/Symfony/Component/Form/Test/DeprecationErrorHandler.php (hunk 3 above)
Comment (carusogabriel) [reply src/Symfony/Component/Form/Test/DeprecationErrorHandler.php]: Gonna revert this one :+1:

**Review by Simperfit — APPROVED** [review]

**Comment by fabpot** [comment]
I'm 👎 here. I don't see the benefit but I can see how we could make a mistake in changing these pieces of code (some changes were wrong in the first iteration after all).

**Pull request closed without merging**